### PR TITLE
Adding sections of code which SIG-Core group owns

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
Any changes to those areas of code will add the o3de/sig-core-maintainers and
o3de/sig-core-reviewers groups to any PRs being merge to the o3de protected branches
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Signed-off-by: lumberyard-employee-dm <56135373+lumberyard-employee-dm@users.noreply.github.com>

### CODE REVIEW

**Review by AMZN-puvvadar — APPROVED** [review]
I see no issues from the sig-network perspective.

**Review by AMZN-alexpete — APPROVED** [review]

**Review by amzn-changml — APPROVED** [review]
LGTM. There are some scripts in the `/cmake/` folder meant for the installer, but sig-build can take ownership for those specific files in a separate PR

**Review by Kadino — APPROVED** [review]
Looks good, thanks for building out this file!

We may want to group directories alphabetically in the future, but I think that can be assessed as this file grows in complexity.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -12,3 +12,28 @@
...
+/Code/CrashHandler/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Code/Framework/AzCore/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Code/Framework/AzFramework/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Code/Framework/AzGameFramework/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Code/Framework/AzToolsFramework/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Gems/CrashReporting/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Gems/LmbrCentral/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Gems/Profiler/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Gems/Prefab/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
+/Gems/ImGui/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
```
Comment (moraaar) [inline .github/CODEOWNERS:27]: Until sig-simulation is created, all the physics gems are owned by sig-core, please add them to the list.

````
/Gems/PhysX/
/Gems/PhysXDebug/
/Gems/NvCloth/
/Gems/Blast/
````

I think it'd be great if automated tests owned by sig-core were also part of this list. Here are the physics ones:

````
/AutomatedTesting/Gem/PythonTests/Physics
/AutomatedTesting/Gem/PythonTests/NvCloth
/AutomatedTesting/Gem/PythonTests/Blast
````

From the sig-core list presented it seems the following automated tests folder should be owned by sig-core too:

````
/AutomatedTesting/Gem/PythonTests/Prefab
/AutomatedTesting/Gem/PythonTests/assetpipeline   // Some subfolders inside assetpipeline might be owned by sig-content, maybe worth clarifying with them.
````

Comment (lumberyard-employee-dm) [reply .github/CODEOWNERS:27]: I can make that change.

**Review by moraaar — APPROVED** [review]

**Pull request merged**